articles/introduction-to-room-db/index.md

```diff
@@ -0,0 +1,155 @@
+
+This article goes through creating an android application that uses room persistent library to create and manage SQLite databases. We are going to create a simple ToDo list application. The list will be stored locally using the library.
+
+Without further ado. Let's dive in!
+
+### Prerequisites
+To follow through with this tutorial, you will need to:
+  1. Have [Android Studio](https://developer.android.com/studio) installed.
+  2. Have a basic knowledge of building Android applications.
+  3. Have a basic understanding of [Kotlin](/engineering-education/kotlin-collections/) programming language, 
+  [SQL](https://en.wikipedia.org/wiki/SQL), 
+  [MVVM](/engineering-education/implementing-mvvm-architecture-in-android-using-kotlin) architecture and Kotlin [coroutines](https://kotlinlang.org/docs/reference/coroutines-overview.html).
+
+Let's get started!
+
+### Step 1 — Getting started
+In this step, we are going to take a look at what makes up the room database. The main components are:
+ 1. Entity - An entity is a class that is annotated with the `@Entity` annotation. This class represents a database table.
+ 2. DAO - A `Data Access Object` is used to map SQL queries to functions. It is an interface annotated with the `@DAO` annotation.
+ 3. Room Database - This class acts as an access point to the SQL database. The class uses the DAO to issue queries to the SQL database.
+
+With that bit of information, let's start implementing room in our application. First, download the starter application from [github](https://github.com/kayere/To-do/tree/start-up). It is just a basic android application with only the UI logic. Our work will be to build the functionality. After downloading, sync the project to download the necessary dependencies.
+
+With that done, let's now proceed to the main task.
+
+### Step 2 — Creating the Entity
+Create a package named `db` and add a Kotlin class inside it. As mentioned earlier, this is the class that we will use to define our table, i.e, its name and column properties.
+
+Add the following code to the class.
+```Kotlin
+@Entity
+data class ToDo( @PrimaryKey val content: String)
+```
+Since we are going to record a sentence for each to-do item, we only need one field in the table. We annotate the field with `@PrimaryKey` to declare it the primary key for our table. There are many annotations for table fields. You can read more about the fields [here](https://developer.android.com/training/data-storage/room/defining-data)
```

Comment:
You could add information about the autogenerating `id`'s in Room. I believe it is an important point for beginners in Room